bluepyentity/entity_definitions.py

```diff
@@ -0,0 +1,338 @@
+
+        if to_add["distribution"] is not None:
+            for item in to_add["distribution"]:
+                resource.add_distribution(**item)
+
+        for file_key in single_files:
+            if to_add[file_key] is not None:
+                setattr(resource, file_key, forge.attach(path=to_add[file_key]))
+
+        return resource
+
+    def get_formatted_definition(self, forge):
+        """Return the formatted definition."""
+        res = {k: v for k, v in self.__dict__.items() if v is not None}
+
+        for k, v in res.items():
+            if hasattr(v, "get_formatted_definition"):
+                res[k] = v.get_formatted_definition(forge)
+
+        return res
+
+
+class ID(BaseModel):
+
+    ids: ListOfStr
+
+    def get_formatted_definition(self, forge):
+        def _wrap_id_fetch(id_):
+            return {
+                "id": id_,
+                "type": _fetch(id_, forge).type,
+            }
+
+        ids = [_wrap_id_fetch(id_) for id_ in self.ids]
+
+        return ids[0] if len(ids) == 1 else ids
+
+
+class IDConverter(Converter):
+    @classmethod
+    def _custom_validator(cls, item):
+        return ID.from_dict({"ids": item})
+
+
+class EntityMixIn(BaseModel):
+
+    wasAttributedTo: IDConverter = None
+    wasGeneratedBy: IDConverter = None
+    wasDerivedFrom: IDConverter = None
+    dateCreated: datetime.datetime = None
+
+
+class Entity(EntityMixIn):
+
+    id: str = None
+    name: str = None
+    description: str = None
+    distribution: DistributionConverter = None
+
+
+class Activity(BaseModel):
+
+    name: str = None
+    status: Literal["Pending", "Running", "Done", "Failed"] = None
+    used: IDConverter = None
+    generated: IDConverter = None
+    startedAtTime: str = None
+    endedAtTime: str = None
+    wasStartedBy: IDConverter = None
+    wasInformedBy: IDConverter = None
+    wasInfluencedBy: IDConverter = None
+
+
+class AnalysisReport(Entity):
+    image: ListOfStr = None
+
+    # Added
+    derivation: str = None
+    types: ListOfStr = None
+
+
+class BrainRegion(BaseModel):
+    id: str
+
+    def get_formatted_definition(self, forge):
+        return forge.reshape(_fetch(self.id, forge), ["id", "label", "notation"])
+
+
+class BrainRegionConverter(Converter):
+    @classmethod
+    def _custom_validator(cls, item):
+        if isinstance(item, str):
+            item = {"id": item}
+        if isinstance(item, dict):
+            return BrainRegion.from_dict(item)
+
+        raise TypeError("str or dict type expected")
+
+
+class BrainLocation(BaseModel):
+    brainRegion: BrainRegionConverter
+
+
+class ModelInstance(Entity):
+    modelOf: str = None
+    brainLocation: BrainLocation = None
+    subject: IDConverter = None
+
+
+class DetailedCircuit(ModelInstance):
+    circuitConfigPath: DataDownload
+    circuitType: str = None
+
+    # These should be removed?
+    circuitBase: str = None
+    nodeCollection: str = None
+    edgeCollection: str = None
+    target: str = None
```

Comment:
If everyone else agrees, I would say yes.

Reply:
Yeah, if we still need them, some fields may need special handling. If not, let's just remove them.